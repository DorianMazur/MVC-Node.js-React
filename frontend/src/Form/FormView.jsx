import React from 'react';
import { Button, Form, Grid, Header } from 'semantic-ui-react';
import FormInputs from './FormInputs.jsx';
import FormMessage from './FormMessage.jsx';

export default class FormView extends React.Component {
  constructor(props) {
    super(props);
    this.state = { showMessage: false, errorMessage: null, firstName: '', lastName: '', email: '' };
    this.updateData = this.updateData.bind(this);
    this.register = this.register.bind(this);
  }

  updateData(data) {
    this.setState({ showMessage: false, errorMessage: null });
    this.setState(data);
  }

  register() {
    const { firstName, lastName, email } = this.state;
    if (!firstName.length || !lastName.length || !email.length) {
      this.setState({ showMessage: true, errorMessage: 'Fill all fields!' });
      return;
    }
    if (!email.match(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i)) {
      this.setState({ showMessage: true, errorMessage: 'Email is wrong!' });
      return;
    }
  }

  render() {
    const { showMessage, firstName, lastName, email, errorMessage } = this.state;
    return (
      <Grid textAlign="center" style={{ height: '100vh' }} verticalAlign="middle">
        <Grid.Column style={{ maxWidth: 450 }}>
          <Header as="h2" textAlign="center">
            Register new account
          </Header>
          <Form size="large">
            <FormInputs updateData={this.updateData} firstName={firstName} lastName={lastName} email={email} />
            <Button fluid size="large" onClick={this.register}>
              Register
            </Button>
          </Form>
          <FormMessage show={showMessage} message={errorMessage} />
        </Grid.Column>
      </Grid>
    );
  }
}
