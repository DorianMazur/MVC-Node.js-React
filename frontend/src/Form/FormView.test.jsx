import '@testing-library/jest-dom';
import React from 'react';

import { render } from '@testing-library/react';
import FormView from './FormView.jsx';

test('FormView render correctly', () => {
  const container = render(<FormView />);
  expect(container.baseElement).toMatchSnapshot();
});
