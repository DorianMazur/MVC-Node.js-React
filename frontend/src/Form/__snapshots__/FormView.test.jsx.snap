// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormView render correctly 1`] = `
<body>
  <div>
    <div
      class="ui center aligned middle aligned grid"
      style="height: 100vh;"
    >
      <div
        class="column"
        style="max-width: 450px;"
      >
        <h2
          class="ui center aligned header"
        >
          Register new account
        </h2>
        <form
          class="ui large form"
        >
          <div
            class="ui segment"
          >
            <div
              class="field"
            >
              <div
                class="ui fluid left icon input"
              >
                <input
                  id="firstName"
                  placeholder="First Name"
                  type="text"
                  value=""
                />
                <i
                  aria-hidden="true"
                  class="user icon"
                />
              </div>
            </div>
            <div
              class="field"
            >
              <div
                class="ui fluid left icon input"
              >
                <input
                  id="lastName"
                  placeholder="Last Name"
                  type="text"
                  value=""
                />
                <i
                  aria-hidden="true"
                  class="user outline icon"
                />
              </div>
            </div>
            <div
              class="field"
            >
              <div
                class="ui fluid left icon input"
              >
                <input
                  id="email"
                  placeholder="Email"
                  type="text"
                  value=""
                />
                <i
                  aria-hidden="true"
                  class="mail icon"
                />
              </div>
            </div>
          </div>
          <button
            class="ui large fluid button"
          >
            Register
          </button>
        </form>
      </div>
    </div>
  </div>
</body>
`;
